src/main/java/com/amazon/opendistroforelasticsearch/ad/rest/handler/ValidateAnomalyDetectorActionHandler.java

```diff
@@ -0,0 +1,590 @@
+    protected static final long HISTORICAL_CHECK_IN_MS = 7776000000L;
+    protected static final String NAME_REGEX = "[a-zA-Z0-9._-]+";
+    protected static final double INTERVAL_RECOMMENDATION_MULTIPLIER = 1.2;
+
+    private final AnomalyDetectionIndices anomalyDetectionIndices;
+    private final AnomalyDetector anomalyDetector;
+
+    private final Logger logger = LogManager.getLogger(IndexAnomalyDetectorActionHandler.class);
+    private final Integer maxAnomalyDetectors;
+    private final Integer maxAnomalyFeatures;
+    private final TimeValue requestTimeout;
+    private final NamedXContentRegistry xContent;
+
+    private ValidateResponse responseValidate;
+    private Map<String, List<String>> failuresMap;
+    private Map<String, List<String>> suggestedChangesMap;
+    private Boolean inferringInterval;
+    private AtomicBoolean inferAgain;
+
+    /**
+     * Constructor function.
+     *
+     * @param client                  ES node client that executes actions on the local node
+     * @param channel                 ES channel used to construct bytes / builder based outputs, and send responses
+     * @param anomalyDetectionIndices anomaly detector index manager
+     * @param anomalyDetector         anomaly detector instance
+     */
+    public ValidateAnomalyDetectorActionHandler(
+        NodeClient client,
+        RestChannel channel,
+        AnomalyDetectionIndices anomalyDetectionIndices,
+        AnomalyDetector anomalyDetector,
+        Integer maxAnomalyDetectors,
+        Integer maxAnomalyFeatures,
+        TimeValue requestTimeout,
+        NamedXContentRegistry xContentRegistry
+    ) {
+        super(client, channel);
+        this.anomalyDetectionIndices = anomalyDetectionIndices;
+        this.anomalyDetector = anomalyDetector;
+        this.maxAnomalyDetectors = maxAnomalyDetectors;
+        this.maxAnomalyFeatures = maxAnomalyFeatures;
+        this.requestTimeout = requestTimeout;
+        this.responseValidate = new ValidateResponse();
+        this.xContent = xContentRegistry;
+        this.inferringInterval = false;
+        this.inferAgain = new AtomicBoolean(true);
+        this.failuresMap = new HashMap<>();
+        this.suggestedChangesMap = new HashMap<>();
+    }
+
+    /**
+     * Start function to process validate anomaly detector request.
+     * Checks if anomaly detector index exist first, if not, add it as a failure case.
+     *
+     * @throws IOException IOException from {@link AnomalyDetectionIndices#initAnomalyDetectorIndexIfAbsent(ActionListener)}
+     */
+    public void startValidation() throws IOException {
+        if (!anomalyDetectionIndices.doesAnomalyDetectorIndexExist()) {
+            anomalyDetectionIndices
+                .initAnomalyDetectorIndex(
+                    ActionListener.wrap(response -> onCreateMappingsResponse(response), exception -> onFailure(exception))
+                );
+        } else {
```

Comment:
why do we need to create detector index if it doesn't exist? I don't think it is in the scope of detector validation. 

Instead, I think we can pass a flag indicating detector index exists or not to next validation method so that the following process can skip validation like duplicate name

Reply:
fixed